common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java

```diff
@@ -144,6 +144,9 @@ private void publish(byte[] buffer, int offset, int length) throws IOException {
         } catch (ExecutionException e) {
             complete(e.getCause());
             throw new IOException(e.getCause());
+        } catch (IllegalStateException e) {
```

Comment:
One minor comment.  Clearly your changes to `onNext()` now throw `IllegalStateException`.  Does anything else in this "wide" `try` block also throw `IllegalStateException`?  If so, is it OK to handle all such `IllegalStateExceptions` here in this manner?  Or should this `IllegalStateException` part of the `catch` block be "scoped tighter" around just the `onNext` call?  I don't know the answer.  I am just raising the point.

Reply:
It's a good question. I haven't seen other places, but I will look a bit more closely.

Reply:
I've now verified that just `onNext` can throw that exception.